src/lib/tls/tls_extensions.h

```diff
@@ -622,9 +633,11 @@ class BOTAN_UNSTABLE_API PSK final : public Extension {
        * @param session_to_resume  the session to be resumed; note that the
        *                           master secret will be taken away from the
        *                           session object.
+       * @param psks               a list of non-resumption PSKs that should be
+       *                           offered to the server
        * @param callbacks          the application's callbacks
        */
-      PSK(Session_with_Handle& session_to_resume, Callbacks& callbacks);
+      PSK(std::optional<Session_with_Handle>& session_to_resume, std::vector<PresharedKey> psks, Callbacks& callbacks);
```

Comment:
Should `session_to_resume` be passed by value or maybe `const&`? Non-`const&` feels like ano out-param? 😕